flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorPartitionLifecycleTest.java

```diff
@@ -102,6 +126,115 @@ public void setup() {
 		rpc = new TestingRpcService();
 	}
 
+	@Test
+	public void testConnectionTerminationAfterExternalRelease() throws IOException, InterruptedException, ExecutionException, TimeoutException {
+		final JobMasterId jobMasterId = JobMasterId.generate();
+
+		final LibraryCacheManager libraryCacheManager = mock(LibraryCacheManager.class);
+		when(libraryCacheManager.getClassLoader(any(JobID.class))).thenReturn(ClassLoader.getSystemClassLoader());
+
+		final JobMasterGateway jobMasterGateway = mock(JobMasterGateway.class);
+		when(jobMasterGateway.getFencingToken()).thenReturn(jobMasterId);
+
+		final TaskManagerActions taskManagerActions = new NoOpTaskManagerActions();
+		final JobManagerConnection jobManagerConnection = new JobManagerConnection(
```

Comment:
We might use `TaskSubmissionTestEnvironment.createJobManagerConnection(jobId, jobMasterGateway, rpc, taskManagerActions, Time.milliseconds(10000L))` instead to avoid mock above `libraryCacheManager` explicitly.